app/src/main/java/com/fpvout/digiview/VideoReaderExoplayer.java

```diff
@@ -26,36 +26,41 @@
 import com.google.android.exoplayer2.source.ProgressiveMediaSource;
 import com.google.android.exoplayer2.upstream.DataSource;
 import com.google.android.exoplayer2.upstream.DataSpec;
-import com.google.android.exoplayer2.util.NonNullApi;
-import com.google.android.exoplayer2.video.VideoListener;
+import com.google.android.exoplayer2.video.VideoSize;
 
 import usb.AndroidUSBInputStream;
 
 public class VideoReaderExoplayer {
     private static final String TAG = "DIGIVIEW";
-    private Handler videoReaderEventListener;
     private SimpleExoPlayer mPlayer;
     static final String VideoPreset = "VideoPreset";
     private final SurfaceView surfaceView;
     private AndroidUSBInputStream inputStream;
-        private UsbMaskConnection mUsbMaskConnection;
+    private UsbMaskConnection mUsbMaskConnection;
     private boolean zoomedIn;
     private final Context context;
     private PerformancePreset performancePreset = PerformancePreset.getPreset(PerformancePreset.PresetType.DEFAULT);
     static final String VideoZoomedIn = "VideoZoomedIn";
     private final SharedPreferences sharedPreferences;
+    private boolean streaming = false;
+
+    private VideoPlayingListener videoPlayingListener = null;
```

Comment:
Listeners could be combined into one interface.

Reply:
yeah, I kept them seperated so you can subscribe to an event in one place and to the other in another place (as you need to overide both methods when you implement it in a combined way )
Or maybe I just wasn't able to do it :p